website/docs/basics/writing-styles.md

```diff
@@ -6,3 +7,122 @@ sidebar_label: Writing Styles
+
+### Never use the actual color name as a variable name
+
+You can name your variables how you like, but we recommend that you never use the actual color name as a variable name. It would be best if you used some generic variable name.
+
+For example, you can call blue color r2d2-color. With this approach, if for any reason your designer chooses to change the project color scheme at any point, you only need to change one variable value, and you are done. 
+
+We like to use [this site](https://chir.ag/projects/name-that-color) for our color names. Also, we recrement using a simple name as possible like:
+
+* Cornflower Blue will be `$base-corn-color` or `$base-flower-color` name.
+
+### All external variables must go to a components map
+
+We use maps for a lot of stuff in our boilerplate. The reason is that:
+
+* maps can contain multiple variables at one place
+* you don't need to write multiple variables at the top.
+* you can nest functionalities
+* you can have multiple values with the same name and just change the top-level map.
+* etc.
+
+We also have many helpers inside our [SASS Docs](/eightshift-docs/sass).
+
+In the component map, you should only add stuff that you will reuse in your component.
+
+**Examples:**
+
+A simple variable in a map.
+
+```scss
+$social-links: (
+  icon-color: global-settings(colors, black),
+);
+
+.social-links {
+  color: map-get-strict($social-links, icon-color);
+}
+```
+
+Nested variables in a map.
+
+```scss
+$social-links: (
+  colors: (
+    dark: global-settings(colors, black),
+    light: global-settings(colors, white),
+  ),
+  sizes: (
+    small: 10px,
+    big: 20px,
+  )
+);
+
+.social-links {
+  color: map-get-deep($social-links, colors, dark);
+
+  &--light {
+    color: map-get-deep($social-links, colors, light);
+  }
+
+  &__sizes {
+    font-size: map-get-deep($social-links, sizes, small);
+
+
+    &--big {
+      font-size: map-get-deep($social-links, sizes, big);
+    }
+  }
+}
+```
+
+### Never mix multiple maps in a single component
+
+You want all your styles to be continued to a single component and not depend on the external stuff. Sometimes this can't be avoided.
+
+With this case, you should  define a key in a component variable for the external map and define it like this:
+
+**Example:**
+
+External map in `assets/styles/parts/utils/_z-index.scss`.
+
+```scss
+$zindex: (
+  header: 100,
+  drawer: 99,
+  overlay: 98,
+);
+```
+
+Your component:
+
+```scss
+$header: (
+  index: map-get-strict($zindex, header),
+);
+
+.header {
+  z-index: map-get-strict($header, index);
+}
+```
+
+### Where can I define global typography for all my blocks/components?
+
+Well, nowhere. Try to move your way of thinking from Bootstrap or any other frontend framework where you define global typography in one place.
+
+Why is that? Because we use component-based development and you should think in that way. Each component is responsible for its styles. 
```

Comment:
```suggestion
Why is that? Because we use component-based development and you should think in that way. Each component is responsible for its own styles. 
```